Example/Example Swift/ComponentAPICoordinator.swift

```diff
@@ -409,15 +409,25 @@ extension ComponentAPICoordinator: CameraViewControllerDelegate {
         validate([document]) { result in
             switch result {
             case .success(let documentRequests):
-                self.documentRequests.append(contentsOf: documentRequests)
-                self.upload(documentRequests: documentRequests)
-                
-                if self.giniConfiguration.multipageEnabled, self.documentRequests.count > 1 {
-                    if let imageDocument = document as? GiniImageDocument {
+                switch document {
+                case let qrCodeDocument as GiniQRCodeDocument:
+                    viewController.showPopup(forQRDetected: qrCodeDocument) {
```

Comment:
Wouldn't it be better to show the popup before the document is validated and definitely before the `cameraDidCapture` method is invoked? That would allow you to keep this code generic and handle all document types the same way.

Reply:
No, because you don't show the popup for not valid QRCodes. If they don't have the right format (bezahle, stuzza...), it shouldn't be shown.